DashAI/back/api/api_v1/endpoints/explainers.py

```diff
@@ -0,0 +1,619 @@
+        database.
+    """
+    with session_factory() as db:
+        try:
+            global_explainer = db.scalars(
+                select(GlobalExplainer).where(GlobalExplainer.id == explainer_id)
+            ).all()
+
+            if not global_explainer:
+                raise HTTPException(
+                    status_code=status.HTTP_404_NOT_FOUND,
+                    detail="Explainer not found",
+                )
+
+            if global_explainer[0].status != ExplainerStatus.FINISHED:
+                raise HTTPException(
+                    status_code=status.HTTP_404_NOT_FOUND,
+                    detail="Explaination not found",
+                )
+
+            explanation_path = global_explainer[0].explanation_path
+
+            with open(explanation_path, "rb") as file:
+                explanation = pickle.load(file)
+
+        except exc.SQLAlchemyError as e:
+            log.exception(e)
+            raise HTTPException(
+                status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+                detail="Internal database error",
+            ) from e
+
+    return explanation
+
+
+@router.get("/global/explanation/plot/{explainer_id}")
+@inject
+async def get_global_explanation_plot(
+    explainer_id: int,
+    component_registry: ComponentRegistry = Depends(
+        Provide[Container.component_registry]
+    ),
+    session_factory: Callable[..., ContextManager[Session]] = Depends(
+        Provide[Container.db.provided.session]
+    ),
+):
+    """Returns the global explanation plot associated with id explainer_id.
+
+    Parameters
+    ----------
+    explaniner_id: int
+        Id to select the global explanation plot to retrieve.
+    session_factory : Callable[..., ContextManager[Session]]
+        A factory that creates a context manager that handles a SQLAlchemy session.
+        The generated session can be used to access and query the database.
+
+    Returns
+    -------
+    List[dict]
+        A JSON with the explanation plot.
+
+    Raises
+    ------
+    HTTPException
+        If there is no global explanation associated with the explanation_id in the
+        database.
+    """
+    with session_factory() as db:
+        try:
+            global_explainer = db.scalars(
+                select(GlobalExplainer).where(GlobalExplainer.id == explainer_id)
+            ).all()
+
+            if not global_explainer:
+                raise HTTPException(
+                    status_code=status.HTTP_404_NOT_FOUND,
+                    detail="Explainer not found",
+                )
+
+            if global_explainer[0].status != ExplainerStatus.FINISHED:
+                raise HTTPException(
+                    status_code=status.HTTP_404_NOT_FOUND,
+                    detail="Explaination plot not found",
+                )
+
+            plot_path = global_explainer[0].plot_path
+
+            with open(plot_path, "rb") as file:
+                plot = pickle.load(file)
```

Comment:
In the same line with the pointed before, it could be possible that some explanation has more than one plot?
And if it so, it is important to this release?